test/e2e/apimachinery/apiserver_identity.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2essh "k8s.io/kubernetes/test/e2e/framework/ssh"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+func getControlPlaneHostname(node *v1.Node) (string, error) {
+	nodeAddresses := e2enode.GetAddresses(node, v1.NodeExternalIP)
+	if len(nodeAddresses) == 0 {
+		return "", errors.New("no valid addresses to use for SSH")
+	}
+
+	controlPlaneAddress := nodeAddresses[0]
+
+	host := controlPlaneAddress + ":" + e2essh.SSHPort
+	result, err := e2essh.SSH("hostname", host, framework.TestContext.Provider)
```

Comment:
Hah, that is one way to do it :)

Reply:
😂

Reply:
apiserver pods generally aren't part of the cluster (static pods managed only by a kubelet)

Reply:
`kubectl debug` has tooling for running a more-privileged Pod on a node to learn about it; maybe we could use some of that approach?

Reply:
just to clarify, I commented because I was curious, we are close to code freeze and I don't mean to delay this PR